### ISSUE TITLE
Errors: add locations

### ISSUE THREAD

**Issue by ghostdogpr** [ghostdogpr/caliban#156]
Add support for the `locations` data for each error as defined in the [GraphQL spec](https://graphql.github.io/graphql-spec/June2018/#sec-Errors):
```json
{
  "errors": [
    {
      "message": "Name for character with ID 1002 could not be fetched.",
      "locations": [ { "line": 6, "column": 7 } ],
      "path": [ "hero", "heroFriends", 1, "name" ],
      "extensions": {
        "code": "CAN_NOT_FETCH_BY_ID",
        "timestamp": "Fri Feb 9 14:33:09 UTC 2018"
      }
    }
  ]
}
```

This will require extracting the line and column information of each field when parsing queries.
See initial discussion in https://github.com/ghostdogpr/caliban/issues/140

**Comment by phderome** [comment]
Clarification request:

I'd like to understand a few basic things about the types involved. I'd like to know whether this is a reasonable extension
 ``` 
 case class ExecutionError(
    msg: String,
    path: List[Either[String, Int]] = Nil,
    locations: List[String] = Nil,
    innerThrowable: Option[Throwable] = None
  ) extends CalibanError
```

I’d like to understand why `path` is of the proper type. it seems that the `Int` part is caliban choosing to specify a List index (from some `reduce` step on Lists that does a zip) whereas I am guessing there is no need to inject additional context for `locations`. From GraphSQL spec, this `locations` is an `array` of `objects`. How would we want to format `"locations": [ { "line": 6, "column": 7 } ]` in the CalibanError `toString` method?

**Comment by phderome** [comment]
How would `GraphQLResponse!handleError` handle the simultaneous presence of `path`, `msg`, and `locations` data? This is more of a behavioural/requirement/modelling question than coding for now.

**Comment by ghostdogpr** [comment]
Closed by #168
